computation_graph/graph.py

```diff
@@ -174,3 +186,28 @@ def source():
         raise NotImplementedError(f"pure source [{name}] should never run")
 
     return make_computation_node(source)
+
+
+@gamla.curry
+def make_terminal(name: str, func: Callable):
+    return base_types.ComputationNode(
+        name=name,
+        func=func,
+        signature=_infer_callable_signature(func),
+        is_terminal=True,
+    )
+
+
+def _aggregator_for_terminal(*args):
+    return tuple(args)
+
+
+DEFAULT_TERMINAL = make_terminal("DEFAULT_TERMINAL", _aggregator_for_terminal)
```

Comment:
i think we can remove `DEFAULT_TERMINAL` and `connect_default_terminal` from the api, they are easy to implement outside